king_phisher/client/windows/campaign_import.py

```diff
@@ -0,0 +1,439 @@
+			return
+		self.select_xml_campaign()
+
+	def signal_multi_close_import_campaign(self, _, __):
+		"""
+		Checks to make sure the import campaign thread is closed before closing.
+		"""
+		if not self.campaign_info:
+			return False
+		if not self.thread_import_campaign:
+			return False
+		if not self.thread_import_campaign.is_alive():
+			return False
+		response = gui_utilities.show_dialog(
+			Gtk.MessageType.QUESTION,
+			'Do you want to cancel importing the campaign?',
+			self.window,
+			message_buttons=Gtk.ButtonsType.YES_NO
+		)
+		print(response)
+		if not response or response == -9:
+			return True
+
+		self.thread_import_campaign.stop()
+		self.thread_import_campaign.join()
+		self.remove_import_campaign()
+
+	def failed_import_action(self):
+		response = gui_utilities.show_dialog(
+			Gtk.MessageType.QUESTION,
+			'Import Failed, do you want to delete this campaign?',
+			self.window,
+			message_buttons=Gtk.ButtonsType.YES_NO
+		)
+		return response
+
+	def remove_import_campaign(self):
+		"""
+		Used to delete the imported campaign on failure or early exit of the import window, if the user
+		selects to have it removed.
+		"""
+		campaign_id = self.campaign_info.find('id').text
+		campaign_name = self.campaign_info.find('name').text
+		campaign_check = self.rpc('db/table/get', 'campaigns', campaign_id)
+		if not campaign_check:
+			return
+		if campaign_name == campaign_check['name']:
+			self.rpc('db/table/delete', 'campaigns', campaign_id)
+			self.logger.info('Deleted Campaign {}'.format(campaign_id))
+
+	def signal_import_button(self, _):
+		"""
+		Function called when the user starts the import campaign process, It will create a new thread,
+		to have the campaign run in the background freeing up the GUI for the user to conduct other functions.
+		"""
+		if not self.campaign_info:
+			self._update_text_view('No campaign information to import')
+			self.button_import_campaign.set_sensitive(False)
+			return
+		self.thread_import_campaign = ImportThread(target=self._import_campaign)
+		self.thread_import_campaign.start()
+
+	def select_xml_campaign(self):
+		"""
+		Prompts the user with a file dialog window to select the King Phisher Campaign XML file to import.
+		Validates the file to make sure it is a Campaign exported from King Phisher and is the correct version to import.
+		"""
+		dialog = extras.FileChooserDialog('Import Campaign from XML', self.window)
+		dialog.quick_add_filter('King Phisher XML Campaign', '*.xml')
+		dialog.quick_add_filter('All Files', '*')
+		response = dialog.run_quick_open()
+		dialog.destroy()
+		if not response:
+			return
+		target_file = response['target_path']
+		self.entry_path.set_text(target_file)
+
+		try:
+			campaign_xml = ET.parse(target_file)
+		except ET.ParseError as error:
+			self.logger.error('cannot import campaign: {0} is not a valid XML file'.format(target_file), error)
+			raise KingPhisherInputValidationError('{0} is not a valid xml file'.format(target_file))
```

Comment:
Strings used as format templates should use double quotes. Same with the line above. This makes it easier to visually identify them from editors that highlight them in different colors.